odl/discr/discr_ops.py

```diff
@@ -101,56 +111,407 @@ def _call(self, x, out=None):
         [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
         """
         return self.inverse
 
+
+class ResizingOperatorBase(Operator):
+
+    """Base class for `ResizingOperator` and `ResizingOperatorAdjoint`.
+
+    This is an abstract class used to share code between the forward and
+    adjoint variants of the resizing operator.
+    """
+
+    def __init__(self, domain, range=None, ran_shp=None, **kwargs):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        domain : uniform `DiscreteLp`
+            Uniformly discretized space, the operator can be applied
+            to its elements.
+        range : uniform `DiscreteLp`, optional
+            Uniformly discretized space in which the result of the
+            application of this operator lies.
+            For the default ``None``, a space with the same attributes
+            as ``domain`` is used, except for its shape, which is set
+            to ``ran_shp``.
+        ran_shp : sequence of int
+            Shape of the range of this operator. This can be provided
+            instead of ``range`` and is mandatory if ``range`` is
+            ``None``.
+        offset : int or sequence of int, optional
+            Number of cells to add to/remove from the left of
+            ``domain.partition``. By default, the difference is
+            distributed evenly, with preference for left in case of
+            ambiguity.
+            This option is can only be used together with ``ran_shp``.
+        pad_mode : str, optional
+            Method to be used to fill in missing values in an enlarged array.
+
+            ``'constant'``: Fill with ``pad_const``.
+
+            ``'symmetric'``: Reflect at the boundaries, not doubling the
+            outmost values. This requires left and right padding sizes
+            to be strictly smaller than the original array shape.
+
+            ``'periodic'``: Fill in values from the other side, keeping
+            the order. This requires left and right padding sizes to be
+            at most as large as the original array shape.
+
+            ``'order0'``: Extend constantly with the outmost values
+            (ensures continuity).
+
+            ``'order1'``: Extend with constant slope (ensures continuity of
+            the first derivative). This requires at least 2 values along
+            each axis where padding is applied.
+
+        pad_const : scalar, optional
+            Value to be used in the ``'constant'`` padding mode.
+
+        discr_kwargs: dict, optional
+            Keyword arguments passed to the `uniform_discr` constructor.
+
+        Examples
+        --------
+        The simplest way of initializing a resizing operator is by
+        providing ``ran_shp`` and, optionally, parameters for the padding
+        variant to be used. The range is inferred from ``domain`` and
+        the supplied parameters. If no ``offset`` is given, the difference
+        in size is evenly distributed to both sides:
```

Comment:
[It is](https://github.com/odlgroup/odl/pull/499/files/b67607feef6f368f8ccac83f2b1eb6335e5137e8#diff-783324f48a9505cd5363ed78952719f9R189)